synse/config.go

```diff
@@ -0,0 +1,26 @@
+package synse
+
+// config.go defines config options for the client.
+
+import (
+	"time"
+)
+
+// Options is the root config options.
+type Options struct {
+	Server ServerOptions
```

Comment:
The reason I used ServerOptions was because:
- I wanted to have some kind of structure for the root config. I thought that Address and Timeout was somewhat related and could belong to a “server” group.
- I was thinking about other server-related config options that could be added in the future once v3 proposal was set, such as transport protocol (http, websocket), security related stuff (tls, role access,…), and so on.

However, now that I think about this again, I find that ServerOptions is not really necessary. These options could totally stand alone by themselves in separate structs. That said, I will get rid of the `ServerOptions` in this PR. More configurable options will be discussed in #3.

Reply:
:+1:

FWIW, `ServerOptions` was not a bad thought and I initially probably would have done the same thing. It was only when I went back through and looked at the example usage that I thought we could simplify. From a flexibility standpoint, the nesting makes sense, but from a usability standpoint we want to make configuring it as simple as we can.